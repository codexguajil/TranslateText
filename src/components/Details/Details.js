import React from 'react'
import PropTypes from 'prop-types'
import { Link, withRouther } from 'react-router-dom'

export const Details = (props) => {
 const {translatedText, original} = props
    return (
        <div>
          <div className='banner'>
          <Link to="/translations" className="header-a">
            Return to Translations
          </Link>
          <Link to="/" className="header-a">
              Translate to French.
          </Link>
        </div>
          <div className="details-field">
            <section className="left">
              <h1>Fran√ßais</h1>
              {translatedText}
            </section>
            <section className="right">
              <h1>English</h1>
              {original}
            </section>
          </div>
        </div>
    )
}

export default Details

Details.propTypes = {
  id: PropTypes.string,
  translatedText: PropTypes.string,
  original: PropTypes.string
}