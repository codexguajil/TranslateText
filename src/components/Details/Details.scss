.details-field {
  background: #FFF;
  color: black;
  display: flex;
  flex-direction: row;
  font-size: 32px;
  height: 600px;
  margin: auto;
  margin-top: 50px;
  width: 95%;
}

.left {
  background: plum;
  height: 500px;
  margin: auto;
  padding: 20px;
  overflow: scroll;
  width: 350px;
}

.right {
  background: cadetblue;
  height: 500px;
  margin: auto;
  padding: 20px;
  overflow: scroll;
  width: 350px;
}