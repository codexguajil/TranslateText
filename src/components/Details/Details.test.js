import React from 'react'
import { shallow } from 'enzyme'
import {Details} from './Details'

describe('Details', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Details />
    )
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})