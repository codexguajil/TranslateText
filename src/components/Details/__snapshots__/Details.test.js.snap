// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Details />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="banner"
        >
          <Link
            className="header-a"
            to="/translations"
          >
            Return to Translations
          </Link>
          <Link
            className="header-a"
            to="/"
          >
            Translate to French.
          </Link>
        </div>,
        <div
          className="details-field"
        >
          <section
            className="left"
          >
            <h1>
              Français
            </h1>
          </section>
          <section
            className="right"
          >
            <h1>
              English
            </h1>
          </section>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              className="header-a"
              to="/translations"
            >
              Return to Translations
            </Link>,
            <Link
              className="header-a"
              to="/"
            >
              Translate to French.
            </Link>,
          ],
          "className": "banner",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Return to Translations",
              "className": "header-a",
              "to": "/translations",
            },
            "ref": null,
            "rendered": "Return to Translations",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Translate to French.",
              "className": "header-a",
              "to": "/",
            },
            "ref": null,
            "rendered": "Translate to French.",
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <section
              className="left"
            >
              <h1>
                Français
              </h1>
            </section>,
            <section
              className="right"
            >
              <h1>
                English
              </h1>
            </section>,
          ],
          "className": "details-field",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1>
                  Français
                </h1>,
                undefined,
              ],
              "className": "left",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Français",
                },
                "ref": null,
                "rendered": "Français",
                "type": "h1",
              },
              undefined,
            ],
            "type": "section",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1>
                  English
                </h1>,
                undefined,
              ],
              "className": "right",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "English",
                },
                "ref": null,
                "rendered": "English",
                "type": "h1",
              },
              undefined,
            ],
            "type": "section",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="banner"
          >
            <Link
              className="header-a"
              to="/translations"
            >
              Return to Translations
            </Link>
            <Link
              className="header-a"
              to="/"
            >
              Translate to French.
            </Link>
          </div>,
          <div
            className="details-field"
          >
            <section
              className="left"
            >
              <h1>
                Français
              </h1>
            </section>
            <section
              className="right"
            >
              <h1>
                English
              </h1>
            </section>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                className="header-a"
                to="/translations"
              >
                Return to Translations
              </Link>,
              <Link
                className="header-a"
                to="/"
              >
                Translate to French.
              </Link>,
            ],
            "className": "banner",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Return to Translations",
                "className": "header-a",
                "to": "/translations",
              },
              "ref": null,
              "rendered": "Return to Translations",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Translate to French.",
                "className": "header-a",
                "to": "/",
              },
              "ref": null,
              "rendered": "Translate to French.",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <section
                className="left"
              >
                <h1>
                  Français
                </h1>
              </section>,
              <section
                className="right"
              >
                <h1>
                  English
                </h1>
              </section>,
            ],
            "className": "details-field",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1>
                    Français
                  </h1>,
                  undefined,
                ],
                "className": "left",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Français",
                  },
                  "ref": null,
                  "rendered": "Français",
                  "type": "h1",
                },
                undefined,
              ],
              "type": "section",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1>
                    English
                  </h1>,
                  undefined,
                ],
                "className": "right",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "English",
                  },
                  "ref": null,
                  "rendered": "English",
                  "type": "h1",
                },
                undefined,
              ],
              "type": "section",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
